core/src/main/java/brooklyn/util/internal/ssh/ShellAbstractTool.java

```diff
@@ -105,7 +105,7 @@ protected static Boolean hasVal(Map<String,?> map, ConfigKey<?> keyC) {
     public static <T> T getOptionalVal(Map<String,?> map, ConfigKey<T> keyC) {
         if (keyC==null) return null;
         String key = keyC.getName();
-        if (map!=null && map.containsKey(key)) {
+        if (map!=null && map.containsKey(key) && map.get(key) != null) {
```

Comment:
`ConfigBag` returns the default if the value is null, so I think it's nice to have the same behaviour here.